IonDotnet/Tree/Impl/IonValue.cs

```diff
@@ -352,6 +354,34 @@ public void WriteTo(IIonWriter writer)
         /// <summary>The <see cref="IonType"/> of this value.</summary>
         public abstract IonType Type { get; }
 
+        // Applicable to IonDecimal
```

Comment:
NotImplementedException probably isn't right here.

I would have a const or static string for the message, saying something like "This operation is not supported for IonType this.Type". Then, maybe an InvalidOperationException which would throw the exception with the message.

Reply:
Done.